packages/plugin-workspace-tools/sources/commands/foreach.ts

```diff
@@ -28,6 +28,30 @@ const getWorkspaceChildrenRecursive = (rootWorkspace: Workspace, project: Projec
   return workspaceList;
 };
 
+/**
+ * Find workspaces marked as dependencies/devDependencies of the current workspace recursively.
+ *
+ * @param rootWorkspace root workspace
+ * @param project project
+ *
+ * @returns all the workspaces marked as dependencies
+ */
+const getWorkspaceDependenciesRecursive = (rootWorkspace: Workspace, project: Project): Array<Workspace> => {
+  const workspaceList = [];
```

Comment:
Should probably use a Set here and you also need to handle circular dependencies so it doesn't get stuck in an infinite loop

Reply:
@merceyz I did a slightly hacky tracking of evaluated dependencies - is what I did clean enough?

Reply:
You seem to have added it to `getWorkspaceChildrenRecursive` instead of `getWorkspaceDependenciesRecursive`

Reply:
@merceyz This is what I get for coding right after waking up.  I've pushed a correction.